subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/AbstractGccCompatibleToolChain.java

```diff
@@ -94,6 +94,8 @@ public AbstractGccCompatibleToolChain(String name, BuildOperationExecutor buildO
 
         target(new Intel32Architecture());
         target(new Intel64Architecture());
+        target(new OsxIntelArchitecture());
```

Comment:
💄 It would be better if we could avoid duplicated architecture. It would be possible to choose different flags by using [getPlatform()](https://github.com/gradle/gradle/blob/5ec3f672ed600a86280be490395d70b7bc634862/subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/DefaultGccPlatformToolChain.java#L80) on the toolchain. The downside of merging `Intel64Architecture` and `OsxIntelArchitecture` is some tests may need to be adjusted which can be troublesome.

Reply:
I think you're suggesting three architectures, correct? And the probing for OS-dependent qualities will happen in their implementations?

- Intel32Architecture
- Intel64Architecture
- Arm64Architecture

Reply:
We could use `Arm64Architecture`, but I think for this PR using `OsxArm64Architecture` would be enough. Adding `arm64` support for all OS is a big can of worm to open.